client/question/components/LearnerViewDirective.js

```diff
@@ -661,6 +661,48 @@ tie.directive('learnerView', [function() {
         var questionWindowDiv =
             document.getElementsByClassName('tie-question-window')[0];
 
+        // Set the name of the hidden property and the change event for
+        // visibility
+        var hidden;
+        var visibilityChange;
+
+        var onVisibilityChange = function() {
+          // When a user changes tabs (or comes back), add a SessionPause
+          // or SessionResumeEvent, respectively.
+          if (document[hidden]) {
+            EventHandlerService.createSessionPauseEvent(
+              SessionIdService.getSessionId());
+          } else {
+            EventHandlerService.createSessionResumeEvent(
+              SessionIdService.getSessionId());
+          }
+        };
+
+        // Code for evaluating and handling page visiblity changes.
+        if (typeof document.hidden !== 'undefined') {
+          // Opera 12.10 and Firefox 18 and later support
+          hidden = 'hidden';
+          visibilityChange = 'visibilitychange';
+        } else if (typeof document.msHidden !== 'undefined') {
+          hidden = 'msHidden';
+          visibilityChange = 'msvisibilitychange';
+        } else if (typeof document.webkitHidden !== 'undefined') {
+          hidden = 'webkitHidden';
+          visibilityChange = 'webkitvisibilitychange';
+        }
+
+        // Warn if the browser doesn't support addEventListener
+        // or the Page Visibility API
+        if (typeof document.addEventListener === 'undefined' ||
+          typeof document[hidden] === 'undefined') {
+          alert('Action not supported by current browser.');
```

Comment:
The user is going to have no idea what's going on when this pops up. We need a more graceful fallback, and we also need to understand the consequences it'll have on our statistics if we can't record certain events -- maybe just don't record pageleave in that case.

Reply:
I think we just won't record them because the EventListener won't be added.